core/app/models/spree/address.rb

```diff
@@ -25,6 +25,8 @@ class Address < Spree::Base
     ADDRESS_FIELDS = %w(firstname lastname company address1 address2 city state zipcode country phone)
     EXCLUDED_KEYS_FOR_COMPARISION = %w(id updated_at created_at deleted_at label user_id)
 
+    default_scope { where(deleted_at: nil) }
```

Comment:
@szymoniwacz let's not use default scopes - they are the root of all evil ;) Let's add this as a scope method in https://github.com/spree/spree/blob/master/api/app/controllers/spree/api/v2/storefront/account/addresses_controller.rb